packages/tentales/src/middleware/middlewares/__tests__/error-middleware.test.ts

```diff
@@ -0,0 +1,215 @@
+/* tslint:disable no-any no-empty */
+
+import Koa from "koa"
+import supertest from "supertest"
+import { errorMiddleware } from "../error-middleware"
+import { createTestMw, createTestMwConfig } from "../../../__test-helpers__"
+import { AuthError } from "../../errors"
+
+describe("Incoming request", () => {
+  test("It does not do anything on the way down", async () => {
+    // Setup
+    const app = new Koa()
+    app.use(errorMiddleware(createTestMwConfig()))
+    app.use(createTestMw())
+    const request = supertest.agent(app.listen())
+
+    // Perform
+    return request.get(`/foo`).expect(200, "Next body")
+  })
+})
+
+describe("Outgoing request", () => {
+  test("It catches runtime errors on the way up", async done => {
+    // Setup
+    const emitStub = jest.fn()
+    const app = new Koa()
+    app.use(
+      createTestMw({
+        fn: ctx => {
+          ctx.app.emit = emitStub
+        },
+      }),
+    )
+    app.use(errorMiddleware(createTestMwConfig()))
+    app.use(
+      createTestMw({
+        fn: ctx => {
+          // tslint:disable-next-line no-unused-expression
+          ctx.apa.boll.not.exists
+        },
+      }),
+    )
+    const request = supertest.agent(app.listen())
+
+    // Perform
+    await request
+      .get(`/foo`)
+      .expect(500)
+      .expect((res: any) => {
```

Comment:
Can't this be simplified? 3 nested expects 😉

Reply:
#wops...